test/unit/toggle/toggle-component.js

```diff
@@ -0,0 +1,151 @@
+import Toggle from '../../../src/components/toggle';
+import ToggleView from '../../../src/views/toggle';
+
+const fakeClient = {
+  fetchCart: () => { return; },
+  createCart: () => { return; },
+};
+
+describe('Toggle Component class', () => {
+  describe('constructor', () => {
+    let toggle;
+    let cart;
+
+    beforeEach(() => {
+      cart = {
+        node: document.createElement('div'),
+      };
+      document.body.appendChild(cart.node);
+    });
+
+    afterEach(() => {
+      document.body.removeChild(cart.node);
+    });
+
+    it('sets typeKey to toggle', () => {
+      toggle = new Toggle({}, {client: fakeClient, cart});
+      assert.equal(toggle.typeKey, 'toggle');
+    });
+
+    it('sets node to node in config if provided', () => {
+      const node = {};
+      toggle = new Toggle({node}, {client: fakeClient, cart});
+      assert.equal(toggle.node, node);
+    });
+
+    it('sets node to a div element before cart node if config does not provide node', () => {
+      const node = document.createElement('div');
+      const div = document.createElement('div');
+      const insertBeforeStub = sinon.stub(Node.prototype, 'insertBefore').returns(node);
+      const createElementStub = sinon.stub(document, 'createElement').returns(div);
+
+      toggle = new Toggle({}, {client: fakeClient, cart});
+      assert.calledOnce(insertBeforeStub);
+      assert.calledWith(insertBeforeStub, div, cart.node);
+      assert.equal(toggle.node, node);
+
+      insertBeforeStub.restore();
+      createElementStub.restore();
+    });
+
+    it('creates a new toggle view', () => {
+      toggle = new Toggle({}, {client: fakeClient, cart});
+      assert.instanceOf(toggle.view, ToggleView);
+    });
+  });
+
+  describe('prototype methods', () => {
+    let toggle;
+    let cart;
+    let toggleVisibilitySpy;
+
+    beforeEach(() => {
+      toggleVisibilitySpy = sinon.spy();
+      cart = {
+        node: document.createElement('div'),
+        toggleVisibility: toggleVisibilitySpy,
+        model: {
+          lineItems: [
+            {
+              id: 1,
+              quantity: 1,
+            },
+            {
+              id: 2,
+              quantity: 2,
+            },
+          ],
+        },
+      };
+      document.body.appendChild(cart.node);
+      toggle = new Toggle({}, {client: fakeClient, cart});
```

Comment:
Similar to above, since not stubbing out the document methods, the constructor's `insertBefore` is going to be adding divs to the actual DOM which aren't cleaned up.